Index: Sources/Subs.php
===================================================================
--- Sources/Subs.php	(revision 776)
+++ Sources/Subs.php	(working copy)
@@ -1180,13 +1180,13 @@
 			$context['sub_template'] = array('main');
 
 		// If we're calling from a page that wants to hide the UI, don't show the sidebar
-		if (empty($context['hide_chrome']))
+		/*if (empty($context['hide_chrome']))
 		{
 			execSubTemplate('sidebar_above', 'ignore');
 			foreach ((array) $context['sidebar_template'] as $key => $template)
 				execSubTemplate($template);
 			execSubTemplate('sidebar_below', 'ignore');
-		}
+		}*/
 
 		execSubTemplate('main_above', 'ignore');
 		// If we're calling from a page that wants to hide the UI, don't show the menus/tabs
Index: Themes/default/index.template.php
===================================================================
--- Themes/default/index.template.php	(revision 776)
+++ Themes/default/index.template.php	(working copy)
@@ -542,7 +542,7 @@
 		</li>';
 	}
 	echo '
-	</ul></div>';
+	</ul><span class="m_moderate wep_modify" id="wep_modify"></span></div>';
 }
 
 // Generate a strip of buttons.
Index: index.php
===================================================================
--- index.php	(revision 776)
+++ index.php	(working copy)
@@ -66,6 +66,7 @@
 require_once($sourcedir . '/Errors.php');
 require_once($sourcedir . '/Load.php');
 require_once($sourcedir . '/Security.php');
+require_once($sourcedir . '/WePortal.php');
 
 // If $maintenance is set specifically to 2, then we're upgrading or something.
 if (!empty($maintenance) && $maintenance == 2)
@@ -155,6 +156,8 @@
 	'viewsmfile' => array('ViewSMFile.php', 'ViewSMFile'),
 	'who' => array('Who.php', 'Who'),
 	'xmlhttp' => array('Xmlhttp.php', 'Xmlhttp'),
+	// WePortal
+	'portal' => array('WePortal.php', 'WePortalAction'),
 );
 
 // Clean the request variables, add slashes, etc.
@@ -312,6 +315,8 @@
 	if (!empty($topic) && ($action !== 'feed') && empty($board_info['cur_topic_approved']) && !allowedTo('approve_posts') && ($user_info['id'] != $board_info['cur_topic_starter'] || $user_info['is_guest']))
 		fatal_lang_error('not_a_topic', false);
 
+	WePortal::instance();
+
 	// Is the forum in maintenance mode? (doesn't apply to administrators.)
 	if (!empty($maintenance) && !allowedTo('admin_forum'))
 	{